src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,762 @@
+		validatePayload(this.payload.length >= 2);
+
+		this._supportedColors = {
+			supportsWarmWhite: Boolean(this.payload[0] & 1),
+			supportsColdWhite: Boolean(this.payload[0] & 2),
+			supportsRed: Boolean(this.payload[0] & 4),
+			supportsGreen: Boolean(this.payload[0] & 8),
+			supportsBlue: Boolean(this.payload[0] & 16),
+			supportsAmber: Boolean(this.payload[0] & 32),
+			supportsCyan: Boolean(this.payload[0] & 64),
+			supportsPurple: Boolean(this.payload[0] & 128),
+			supportsIndex: Boolean(this.payload[1] & 1),
+		};
+
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+
+		for (const key of ColorKeys) {
+			const supportsColorKey = supportedColorTableKey(key);
+
+			const valueId: ValueID = {
+				commandClass: CommandClasses["Color Switch"],
+				endpoint: endpoint.index,
+				property: "supportedColor",
+				propertyKey: key,
+			};
+
+			// TODO: Should we be setting this every run?
+			node.valueDB.setMetadata(valueId, {
+				...ValueMetadata.ReadOnly,
+				label: `Supports ${key}`,
+				description: `Whether the endpoint supports setting the ${key} color.`,
+			});
+			node.valueDB.setValue(
+				valueId,
+				this._supportedColors[supportsColorKey] ?? false,
+			);
+		}
+	}
+
+	private _supportedColors: SupportedColorTable | null;
+
+	public get supportsWarmWhite(): boolean | undefined {
+		return this._supportedColors?.supportsWarmWhite;
+	}
+
+	public get supportsColdWhite(): boolean | undefined {
+		return this._supportedColors?.supportsColdWhite;
+	}
+
+	public get supportsRed(): boolean | undefined {
+		return this._supportedColors?.supportsRed;
+	}
+
+	public get supportsGreen(): boolean | undefined {
+		return this._supportedColors?.supportsGreen;
+	}
+
+	public get supportsBlue(): boolean | undefined {
+		return this._supportedColors?.supportsBlue;
+	}
+
+	public get supportsAmber(): boolean | undefined {
+		return this._supportedColors?.supportsAmber;
+	}
+
+	public get supportsCyan(): boolean | undefined {
+		return this._supportedColors?.supportsCyan;
+	}
+
+	public get supportsPurple(): boolean | undefined {
+		return this._supportedColors?.supportsPurple;
+	}
+
+	public get supportsIndex(): boolean | undefined {
+		return this._supportedColors?.supportsIndex;
+	}
+
+	public toJSON(): JSONObject {
+		return super.toJSONInherited({
+			...this._supportedColors,
+		});
+	}
+}
+
+@CCCommand(ColorSwitchCommand.SupportedGet)
+@expectedCCResponse(ColorSwitchCCSupportedReport)
+export class ColorSwitchCCSupportedGet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | CCCommandOptions,
+	) {
+		super(driver, options);
+	}
+}
+
+@CCCommand(ColorSwitchCommand.Report)
+export class ColorSwitchCCReport extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		if (this.version >= 3) {
+			validatePayload(this.payload.length >= 4);
```

Comment:
Out of consistency, I would move this check into the if branch below (where you read the payload).
Although this would mean a little additional work, the check is logically closer to where its needed.